### PR TITLE
Fix free trials in PayPal Express. #1107

### PR DESCRIPTION
**Pull request by nosegraze**
#1107

### CODE REVIEW

File: includes/gateways/class-rcp-payment-gateway-paypal-express.php [hunk 1]
```diff
@@ -210,11 +214,16 @@ public function process_confirmation() {
...
-					$args['TRIALBILLINGPERIOD']      = ucwords( $this->subscription_data['trial_duration_unit'] );
-					$args['TRIALBILLINGFREQUENCY']   = $this->subscription_data['trial_duration'];
+				if ( $details['is_trial'] ) {
+					$args['TRIALBILLINGPERIOD']      = ucwords( $details['subscription']['trial_duration_unit'] );
+					$args['TRIALBILLINGFREQUENCY']   = $details['subscription']['trial_duration'];
 					$args['TRIALTOTALBILLINGCYCLES'] = 1;
 					$args['TRIALAMT']                = 0;
+
+					// Set profile start date to the end of the free trial.
+					$args['PROFILESTARTDATE'] = date( 'Y-m-d\TH:i:s', strtotime( '+' . $details['subscription']['trial_duration'] . ' ' . $details['subscription']['trial_duration_unit'], current_time( 'timestamp' ) ) );
```
Comment (mindctrl) [inline includes/gateways/class-rcp-payment-gateway-paypal-express.php]: is this time format correct with the `\T`?

Comment (nosegraze) [inline includes/gateways/class-rcp-payment-gateway-paypal-express.php]: I copied it from where it's used further up (also uses `\T`).

Comment (mindctrl) [inline includes/gateways/class-rcp-payment-gateway-paypal-express.php]: Ah ok. I'm not hip on all the date formats in PHP. Looked invalid but it's valid and how PayPal likes it to be. If you space it, you get the full timezone code (UTC), which is how I've always used it, but they don't want that. This looks good to me. Ignore me :D

Comment (pippinsplugins) [inline includes/gateways/class-rcp-payment-gateway-paypal-express.php]: It is correct :+1:

**Pull request merged**